src/kernel/mconfig_dot.ml

```diff
@@ -233,7 +237,13 @@ end
 let prepend_config ~dir:cwd configurator (directives : directive list) config =
   List.fold_left ~init:(config, []) ~f:(fun (config, errors) ->
     function
-    | `B path -> {config with build_path = path :: config.build_path}, errors
+    | `B path ->
+      (* CR ccasinghino: Probably some way to put hidden includes in merlin files is
+         needed. *)
+      let build_path = { config.build_path with
+                         visible = path :: config.build_path.visible }
```

Comment:
I think this means that some editors won't ever add anything to the `hidden` directory.

Reply:
You're saying some editors write merlin files with the includes instead of just passing the flags?  Do you know which editors do this?

It's probably particularly hard to extend the format, I just didn't know merlin config files were used for anything other than per-project configuration.

Reply:
Resolved offline. We'll just need to make a change to our build system.

Reply:
(We concluded in an out of band conversation that this is fine if we can arrange for `-H` to be passed as a `FLG` in the .merlin file, and are working to make that happen.  Probably I should add tests that do it that way, though I don't think that needs to block this PR)